e2e/create-pvc_test.go

```diff
@@ -67,69 +70,71 @@ var _ = Describe("CreatePVC", func() {
 	})
 
 	It("creates a pvc and attaches to pod", Label("pvcCreation"), func() {
+		pvcName := "test-pvc"
+		storageClassName := "kubevirt"
 
-		volumeName := "pv1"
-		image := "busybox"
 		podName := "test-pod"
-		pvcName := "test-pvc"
 		cmd := []string{"sh"}
 		args := []string{"-c", "while true; do ls -la /opt; echo this file system was made availble using kubevirt-csi-driver; mktmp /opt/test-XXXXXX; sleep 1m; done"}
-		storageClassName := "kubevirt"
 
-		quantity, err := resource.ParseQuantity("1Gi")
+		pvc := pvcSpec(pvcName, storageClassName, "1Gi")
+
+		By("creating a pvc")
+		_, err := tenantClient.CoreV1().PersistentVolumeClaims(namespace).Create(context.Background(), pvc, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
-		pvc := &k8sv1.PersistentVolumeClaim{
-			ObjectMeta: metav1.ObjectMeta{Name: pvcName},
-			Spec: k8sv1.PersistentVolumeClaimSpec{
-				AccessModes: []k8sv1.PersistentVolumeAccessMode{k8sv1.ReadWriteOnce},
-				Resources: k8sv1.ResourceRequirements{
-					Requests: k8sv1.ResourceList{
-						"storage": quantity,
-					},
-				},
-				StorageClassName: &storageClassName,
-			},
-		}
+		pod := podWithPvcSpec(podName, pvcName, cmd, args)
+		By("creating a pod that attaches pvc")
+		newPod, err := tenantClient.CoreV1().Pods(namespace).Create(context.Background(), pod, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
 
-		pod := &k8sv1.Pod{
-			ObjectMeta: metav1.ObjectMeta{
-				GenerateName: podName,
-			},
-			Spec: k8sv1.PodSpec{
-				RestartPolicy: k8sv1.RestartPolicyNever,
-				Containers: []k8sv1.Container{
-					{
-						Name:    podName,
-						Image:   image,
-						Command: cmd,
-						Args:    args,
-						VolumeMounts: []k8sv1.VolumeMount{
-							{
-								Name:      volumeName,
-								MountPath: "/opt",
-							},
-						},
-					},
-				},
-				Volumes: []k8sv1.Volume{
-					{
-						Name: volumeName,
-						VolumeSource: k8sv1.VolumeSource{
-							PersistentVolumeClaim: &k8sv1.PersistentVolumeClaimVolumeSource{
-								ClaimName: pvc.GetName(),
-							},
-						},
-					},
-				},
-			},
+		By("Wait for pod to reach a running phase")
+		Eventually(func() error {
+			updatedPod, err := tenantClient.CoreV1().Pods(namespace).Get(context.Background(), newPod.Name, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if updatedPod.Status.Phase != k8sv1.PodRunning {
+				return fmt.Errorf("Pod in phase %s, expected Running", updatedPod.Status.Phase)
+			}
+			return nil
+		}, 3*time.Minute, 5*time.Second).Should(Succeed(), "pod should reach running state")
```

Comment:
we could probably have helper functions for waiting for the pod to hit a running state, and another one for waiting for a pod to be deleted.

Reply:
yes